definitions/npm/detox_v7.0.x/test_detox_v7.0.x.js

```diff
@@ -0,0 +1,170 @@
+/* @flow */
+
+const detox = require("detox");
+const {device, expect, element, by, waitFor} = require("detox");
+
+const configs = {
+    configurations: {
+        "ios.sim.debug": {
+            binaryPath:
+                "packages/apps/Example/ios/build/Build/Products/Debug-iphonesimulator/example.app",
+            build:
+                "xcodebuild -project packages/apps/Example/ios/Example.xcodeproj -scheme Example -configuration Debug -sdk iphonesimulator -derivedDataPath packages/apps/Example/ios/build",
+            type: "ios.simulator",
+            name: "iPhone X"
+        }
+    }
+};
+
+
+// Detox Object API
+// ----------------
+detox.cleanup();
+// $ExpectError
+detox.init({configurations: ''}, {initGlobals: 123, launchApp: false});
+detox.init(configs, {initGlobals: false});
+
+
+// Device Object API
+// -----------------
+device.launchApp({})
+device.launchApp({newInstance: true})
+device.launchApp({permissions: {calendar: 'YES'}});
+device.launchApp({url: 'url', newInstance: false});
+device.launchApp({userNotification: { "title": "push" }, newInstance: true});
+device.launchApp({launchArgs: {arg1: 1, arg2: "2"}});
+// $ExpectError
+device.launchApp({permissions: false, url: false});
+
+// $ExpectError
+device.terminateApp({err: 'err'})
+device.terminateApp()
+device.terminateApp('other.bundle.id')
+device.reloadReactNative()
+device.sendToHome()
+device.installApp()
+device.uninstallApp()
+// $ExpectError
+device.openURL({url: 'https://flow.org', sourceApp: false})
+device.openURL({url: 'https://flow.org'})
+
+// $ExpectError
+device.sendUserNotification()
+device.sendUserNotification({})
+
+// $ExpectError
+device.setOrientation('normal')
+device.setOrientation('portrait')
+
+// $ExpectError
+device.setLocation()
+device.setLocation(32.04, 34.78)
+
+device.setURLBlacklist(['https://flow.org','https://flow.org'])
+device.enableSynchronization()
+device.disableSynchronization()
+device.resetContentAndSettings()
+device.getPlatform()
+
+
+// Matchers
+// --------
+by.id('flowType')
+by.text('flowType is great!')
+by.label('welcome')
+// $ExpectError
+by.label(123)
+by.type('RCTImageView')
+by.traits(['button'])
+// $ExpectError
+by.traits('button')
+
+
+// Actions on element
+// ------------------
+ element(by.id('testElement')).tap()
```

Comment:
The whole API is async (promise based), can flow enforce async usage across the API ?

Reply:
We cannot enforce the using to `away` on these. 

But I think all the `=> {}` and `=> void` should be replace by `=> Promise<void>`.

Reply:
changed return types to `Promise<void>`. 👍